test/suites/drift/suite_test.go

```diff
@@ -337,6 +337,115 @@ var _ = Describe("Drift", Label("AWS"), func() {
+				})
+				g.Expect(len(cordonedNodes)).To(BeNumerically("==", 1))
+			}).Should(Succeed())
+
+			// Drift should fail and original node should be uncordoned
+			// TODO: reduce timeouts when deprovisioning waits are factored out
+			machineList := &v1alpha5.MachineList{}
+			Eventually(func(g Gomega) {
+				g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(cordonedNodes[0]), cordonedNodes[0]))
+				g.Expect(cordonedNodes[0].Spec.Unschedulable).To(BeFalse())
+				g.Expect(env.Client.List(env, machineList, client.HasLabels{test.DiscoveryLabel})).To(Succeed())
+				g.Expect(len(machineList.Items)).To(BeNumerically("==", int(numPods)))
+			}).WithTimeout(20 * time.Minute).Should(Succeed())
+			env.ConsistentlyExpectNodeCount(">=", int(numPods), "20s")
+		})
+		It("should not continue to drift if a node registers but never becomes initialized", func() {
+			// launch a new machine
+			var numPods int32 = 2
+			dep := test.Deployment(test.DeploymentOptions{
+				Replicas: 2,
+				PodOptions: test.PodOptions{
+					ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "inflate"}},
+					PodAntiRequirements: []v1.PodAffinityTerm{{
+						TopologyKey: v1.LabelHostname,
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{"app": "inflate"},
+						}},
+					},
+				},
+			})
+			env.ExpectCreated(dep, nodeTemplate, provisioner)
+			env.EventuallyExpectNodeCount("==", int(numPods))
+
+			// Drift the machine with bad configuration
+			provisioner.Spec.StartupTaints = []v1.Taint{{Key: "example.com/taint", Effect: v1.TaintEffectPreferNoSchedule}}
+			env.ExpectCreatedOrUpdated(provisioner)
+
+			// Should see the machine has drifted
+			machines := env.EventuallyExpectCreatedMachineCount("==", int(numPods))
+			Eventually(func(g Gomega) {
+				for _, machine := range machines {
+					g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(machine), machine)).To(Succeed())
+					g.Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted).IsTrue()).To(BeTrue())
+				}
+			}).Should(Succeed())
+
+			// Expect nodes To be cordoned
+			cordonedNodes := []*v1.Node{}
+			nodes := env.EventuallyExpectNodeCount("==", 3)
+			Eventually(func(g Gomega) {
+				cordonedNodes = lo.Filter(nodes, func(node *v1.Node, _ int) bool {
+					g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(node), node))
+					return node.Spec.Unschedulable == true
+				})
+				g.Expect(len(cordonedNodes)).To(BeNumerically("==", 1))
+			}).Should(Succeed())
+
+			// Drift should fail and original node should be uncordoned
+			// TODO: reduce timeouts when deprovisioning waits are factored out
+			Eventually(func(g Gomega) {
+				g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(cordonedNodes[0]), cordonedNodes[0]))
+				g.Expect(cordonedNodes[0].Spec.Unschedulable).To(BeFalse())
+			}).WithTimeout(15 * time.Minute).Should(Succeed())
+			env.ConsistentlyExpectNodeCount("==", 3, "20s")
```

Comment:
Same comments as on the other test. We should be validating the Machine count, not the node count to ensure that we are not continuing to provision or scale-down.